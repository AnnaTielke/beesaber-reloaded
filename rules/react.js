module.exports = {
  plugins: ['react'],

  extends: ['plugin:react/recommended'],

  settings: {
    react: {
      version: 'detect',
    },
  },

  rules: {
    'react/button-has-type': 'error',
    'react/jsx-boolean-value': 'error',
    'react/jsx-fragments': 'error',
    'react/jsx-handler-names': 'error',
    'react/jsx-no-bind': ['error', { allowArrowFunctions: true }],
    'react/jsx-pascal-case': 'error',
    'react/no-access-state-in-setstate': 'error',
    'react/no-array-index-key': 'error',
    'react/no-danger': 'error',
    'react/no-did-mount-set-state': 'error',
    'react/no-did-update-set-state': 'error',
    'react/no-multi-comp': 'error',
    'react/no-this-in-sfc': 'error',
    'react/no-typos': 'error',
    'react/no-unsafe': 'error',
    'react/no-unused-prop-types': 'error',
    'react/no-unused-state': 'error',
    'react/no-will-update-set-state': 'error',
    'react/prefer-es6-class': 'error',
    'react/prefer-stateless-function': 'error',
    'react/style-prop-object': 'error',
    'react/void-dom-elements-no-children': 'error',
  },
};
