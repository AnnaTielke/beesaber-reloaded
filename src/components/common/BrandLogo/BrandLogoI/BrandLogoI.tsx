import React from 'react';
import styled from 'styled-components';

import tokens from '../../../../tokens/index';

const yellow = tokens.color.yellow;

const Container = styled.div`
  font-family: 'NeonTubes2';
`;

const IStroke = styled.span`
  color: ${yellow.light};
  text-shadow: ${yellow.main} -2px -1px 5px, ${yellow.main} 2px 1px 5px,
    ${yellow.main} 0px 0px 10px, ${yellow.main} 0px 0px 15px,
    ${yellow.main} 0px 0px 20px, ${yellow.main} 0px 0px 30px;
`;

const BrandLogoI = (): JSX.Element => (
  <Container>
    <IStroke>I</IStroke>
  </Container>
);

export default BrandLogoI;
