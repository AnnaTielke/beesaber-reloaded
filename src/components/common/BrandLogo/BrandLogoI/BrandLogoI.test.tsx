import { render } from '@testing-library/react';
import React from 'react';

import BrandLogoI from './BrandLogoI';

describe('BrandLogoI', () => {
  it('should match the snapshot', () => {
    const { container } = render(<BrandLogoI />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
