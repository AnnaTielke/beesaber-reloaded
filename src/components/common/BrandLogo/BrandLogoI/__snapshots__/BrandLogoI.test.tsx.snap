// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandLogoI should match the snapshot 1`] = `
<div
  class="sc-bdvvtL etKHLZ"
>
  <span
    class="sc-gsDKAQ fDJHTz"
  >
    I
  </span>
</div>
`;
